server/protocols/webadmin/webadmin-mailqueue/src/test/java/org/apache/james/webadmin/routes/MailQueueRoutesTest.java

```diff
@@ -355,39 +364,56 @@ public void deleteMailsShouldReturnBadRequestWhenRecipientIsInvalid() {
     }
 
     @Test
-    public void deleteMailsShouldReturnNoContentWhenSenderIsValid() {
+    public void deleteMailsTaskShouldCompleteWhenSenderIsValid() throws Exception{
         mailQueueFactory.createQueue(FIRST_QUEUE);
 
-        given()
-            .param("sender", "sender@james.org")
-        .when()
+        String taskId = with()
+            .param("sender", SENDER_1_JAMES_ORG)
             .delete(FIRST_QUEUE + "/mails")
-        .then()
-            .statusCode(HttpStatus.NO_CONTENT_204);
+            .jsonPath()
+                .getString("taskId");
+
+        taskShouldBeCompleted(taskId);
     }
 
     @Test
     public void deleteMailsShouldReturnNoContentWhenNameIsValid() {
         mailQueueFactory.createQueue(FIRST_QUEUE);
 
-        given()
+        String taskId = with()
             .param("name", "mailName")
-        .when()
             .delete(FIRST_QUEUE + "/mails")
-        .then()
-            .statusCode(HttpStatus.NO_CONTENT_204);
+            .jsonPath()
+                .getString("taskId");
+
+        taskShouldBeCompleted(taskId);
     }
 
     @Test
     public void deleteMailsShouldReturnNoContentWhenRecipientIsValid() {
         mailQueueFactory.createQueue(FIRST_QUEUE);
 
+        String taskId = with()
+                .param("recipient", RECIPIENT_JAMES_ORG)
+                .delete(FIRST_QUEUE + "/mails")
+                .jsonPath()
+                .getString("taskId");
+
+        taskShouldBeCompleted(taskId);
+    }
+
+    private void taskShouldBeCompleted(String taskId) {
```

Comment:
We avoid method extraction on assertions to avoid navigating in test (this aims at making comprehension easier...)

You should not rely on that method.